.github/workflows/style.yaml

```
@@ -25,12 +25,35 @@ jobs:
         uses: actions/checkout@v2
         with:
           path: ${{ github.event.repository.name }}
+          fetch-depth: 0
 
       - name: Install styler 👚
         run: |
           remotes::install_github("r-lib/styler", auth_token = "${{ secrets.GITHUB_TOKEN }}")
         shell: Rscript {0}
 
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@v17
+        with:
+          path: ${{ github.event.repository.name }}
+          separator: ","
+          files: |
+            **.R
+            **.Rmd
+            **.Rnw
+
       - name: Run styler 👟
-        run: styler::style_pkg("${{ github.event.repository.name }}", dry = "fail")
+        run: |
+          options(warn=-1)
+          changed_files <- unlist(strsplit(
+            "${{ steps.changed-files.outputs.all_changed_files }}",
+            split=","
+          ))
+          for (file in changed_files) {
+            ext <- tools::file_ext(file)
+            if (ext == "R" || ext == "Rmd" || ext == "Rnw") {
+              styler::style_file(file, dry = "fail")
+            }
+          }
```

@pawelru @kpagacz @nikolas-burkoff - Needed your help here. Do y'all know why I can't get this to fail? Styler is recognizing that the file needs to be styled but the exit code is not causing the step to fail 🤔

---

That looks good. `options` setting was my first candidate to check but it has no effect... Styler generates a "rlang-type-of-error" (in particular: not a standard/base error) and maybe there is some weird error handling within a for loop. I would give a try and convert back to a single function call
```suggestion
          options(warn=-1)
          changed_files <- unlist(strsplit(
            "${{ steps.changed-files.outputs.all_changed_files }}",
            split=","
          ))
          is_r_file <- function(x) {
              ext <- tools::file_ext(x)
              ext == "R" || ext == "Rmd" || ext == "Rnw"
          }
          changed_r_files <- Filter(is_r_file, changed_files)
          styler::style_file(changed_r_files, dry = "fail")
```